### ISSUE TITLE
ArgumentError: bad value for range on accessing range with infinite lower range

### ISSUE THREAD

**Issue by liaden** [rails/rails#39833]
### Steps to reproduce
1. Create a `Thing` table with a `tsrange` column named `active_during`.
```
class CreateThings < ActiveRecord::Migration[6.0]
  def change
    create_table :things do |t|
      t.tsrange :active_during, null: false
      t.timestamps
    end
  end
end
```
2. Create an instance of `Thing`: `Thing.create(active_during: nil..Date.tomorrow)`.
3. Access the tsrange column on the instance after retrieving from the DB: `Thing.last.active_during`

### Expected behavior
`Thing.last.active_during` results in the same behavior as `Thing.new(active_during: nil..Date.tomorrow)`: `nil..Tue, 14 Jul 2020`. 

Note: this works fine on infinite upper ranges.

### Actual behavior
`ArgumentError: bad value for range`

### System configuration
**Rails version**:
6.0.3.2

**Ruby version**:
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin18]

**Comment by iSarCasm** [comment]
@composerinteralia This can be resolved by a Float monkey-patch (maybe as part of ActiveSupport):
```
class Float
  alias_method :original_compare, :<=>
  def <=>(other)
    if other.is_a? Time
      -(other <=> self)
    else
      original_compare(other)
    end
  end
end
```
But I don't how I feel about that. As of Ruby 2.7 it already supports beginless/endless Time and Date ranges. Date also supports ranges from `-Infinity` but Time DOES NOT (as you mentioned).

An alternative to monkey-patching would be updating Ruby itself to support `-Float::INFINITY..Time.new` (as well as `Time.new..Float::INFINITY`) on native level

Another alternative would be changing ActiveRecord `cast_value` behavior from using `Float::INFINITY` to using `nil` instead? That would require ActiveRecord ruby version bump from `2.5` to `2.7` which is huge. 

What if we allow `2.7` users to use beginless/endless ranges with `cast_value` using `nil` and version below default to `Infinity`?

Thoughts?

**Comment by Mr0grog** [comment]
I ran into this today as well. The solution above that monkey-patches float seems extremely convenient, but I also totally understand the concerns with doing that. (The degree to which ActiveSupport and many Ruby gems monkey-patch built-in stuff really gave me pause when I first picked up both Ruby and Rails.)

I’d be happy with any solution that makes these values accessible, whether it’s 3.ii (3.i seems like imposing an unreasonable upgrade burden on lots of Rails users), a custom range type for this situation (then we could also support ranges that exclude the start value), or even if ActiveRecord just gave me back the Postgres string representation (`'[-infinity,2020-08-22 00:00:00]'`) and also let me set it using the string. (This last one is definitely the worst solution, but it’s still better than not being able to read a value like this at all.)
